lib/storage/metadata/in_memory/bucket_utilities.js

```diff
@@ -6,9 +6,9 @@ function markerFilterMPU(allMarkers, array) {
         // as or alphabetically after the uploadId of the item,
         // eliminate the item.
         if (uploadIdMarker && keyMarker === array[i].key) {
```

Comment:
The logic of the whole function seems quite convoluted for what it does, what about going to a simpler:

```javascript
const firstUnfilteredIndex = array.findIndex(
    item => (uploadIdMarker && item.key === keyMarker ?
             item.uploadId > uploadIdMarker :
             item.key > keyMarker));
    return firstUnfilteredIndex !== -1 ? array.slice(firstUnfilteredIndex) : array;
```

Reply:
nice!